Bug report: Client scripts are not runnable

**Environment (dev/staging/live, version)**

`master` branch at 17cbccf, but I know for sure it has existed since some time ago (to be explained later).

**Steps to Reproduce**

Run any client script that connects with the Datastore.

**Expected Behaviour**

The client script is runnable, or any failures should be attributed to the script's obsolescence.

**Actual Behaviour**

The following exception is shown on the console (example when running `DataMigrationForTimezoneFieldInCourses`):

```
Exception in thread "main" java.lang.ExceptionInInitializerError
	at teammates.logic.core.AccountsLogic.<clinit>(AccountsLogic.java:32)
	at teammates.logic.api.Logic.<clinit>(Logic.java:73)
	at teammates.client.scripts.DataMigrationForTimezoneFieldInCourses.<clinit>(DataMigrationForTimezoneFieldInCourses.java:66)
Caused by: javax.jdo.JDOFatalUserException: Application code attempted to create a PersistenceManagerFactory named transactions-optional, but one with this name already exists!  Instances of PersistenceManagerFactory are extremely slow to create and it is usually not necessary to create one with a given name more than once.  Instead, create a singleton and share it throughout your code.  If you really do need to create a duplicate PersistenceManagerFactory (such as for a unittest suite), set the appengine.orm.disable.duplicate.pmf.exception system property to avoid this error.
	at org.datanucleus.store.appengine.jdo.DatastoreJDOPersistenceManagerFactory.getPersistenceManagerFactory(DatastoreJDOPersistenceManagerFactory.java:135)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at javax.jdo.JDOHelper$16.run(JDOHelper.java:1956)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.jdo.JDOHelper.invoke(JDOHelper.java:1951)
	at javax.jdo.JDOHelper.invokeGetPersistenceManagerFactoryOnImplementation(JDOHelper.java:1159)
	at javax.jdo.JDOHelper.getPersistenceManagerFactory(JDOHelper.java:803)
	at javax.jdo.JDOHelper.getPersistenceManagerFactory(JDOHelper.java:1086)
	at javax.jdo.JDOHelper.getPersistenceManagerFactory(JDOHelper.java:914)
	at teammates.storage.api.EntitiesDb.<clinit>(EntitiesDb.java:51)
	... 3 more
```

**Cause**

This is due to #6240. Previously, the `PersistenceManagerFactory` is hidden inside `Datastore` package and accessed by an accessor method. Now it is being duplicated in `EntitiesDb` and `RemoteApiClient`, and as it turns out, instantiating more than one such object is disallowed.

--- a/src/client/java/teammates/client/remoteapi/RemoteApiClient.java
+++ b/src/client/java/teammates/client/remoteapi/RemoteApiClient.java
@@ -1,10 +1,12 @@
 package teammates.client.remoteapi;
 
 import java.io.IOException;
+import java.lang.reflect.Method;
 
-import javax.jdo.JDOHelper;
 import javax.jdo.PersistenceManager;
 
+import teammates.storage.api.CoursesDb;
+import teammates.storage.api.EntitiesDb;
 import teammates.test.driver.TestProperties;
 
 import com.google.appengine.tools.remoteapi.RemoteApiInstaller;
@@ -12,10 +14,22 @@ import com.google.appengine.tools.remoteapi.RemoteApiOptions;
 
 public abstract class RemoteApiClient {
     
-    protected static final PersistenceManager PM =
-            JDOHelper.getPersistenceManagerFactory("transactions-optional").getPersistenceManager();
+    protected static final PersistenceManager PM = getPm();
     
     private static final String LOCALHOST = "localhost";
+    
+    private static PersistenceManager getPm() {
+        try {
+            // use reflection to bypass the visibility level of the method
+            Method method = EntitiesDb.class.getDeclaredMethod("getPm");
+            method.setAccessible(true);
+            
+            // the method is non-static and EntitiesDb is an abstract class; use any *Db to invoke it
+            return (PersistenceManager) method.invoke(new CoursesDb());
+        } catch (ReflectiveOperationException e) {
+            throw new RuntimeException(e);
+        }
+    }
     
     protected void doOperationRemotely() throws IOException {
 

